subliminal/providers/itasa.py

```diff
@@ -0,0 +1,390 @@
+
+        :param str series: series of the episode.
+        :return: the show id, if found.
+        :rtype: int or None
+
+        """
+        series_sanitized = sanitize(series).lower()
+        show_ids = self._get_show_ids()
+        show_id = None
+
+        # attempt with country
+        if not show_id and country_code:
+            logger.debug('Getting show id with country')
+            show_id = show_ids.get('%s %s' % (series_sanitized, country_code.lower()))
+
+        # attempt clean
+        if not show_id:
+            logger.debug('Getting show id')
+            show_id = show_ids.get(series_sanitized)
+
+        # search as last resort
+        if not show_id:
+            logger.warning('Series not found in show ids')
+            show_id = self._search_show_id(series)
+
+        return show_id
+
+    def _download_zip(self, sub_id):
+        # download the subtitle
+        logger.info('Downloading subtitle %r', sub_id)
+
+        params = {
+            'authcode': self.auth_code,
+            'apikey': self.apikey,
+            'subtitle_id': sub_id
+        }
+
+        r = self.session.get(self.server_url + 'subtitles/download', params=params, timeout=30)
+        r.raise_for_status()
+
+        return r.content
+
+    def query(self, series, season, episode, format, resolution, country=None, hash=None):
+
+        # To make queries you need to be logged in
+        if not self.logged_in:
+            raise ConfigurationError('Cannot query if not logged in')
+
+        # get the show id
+        show_id = self.get_show_id(series, country)
+        if show_id is None:
+            logger.error('No show id found for %r ', series)
+            return []
+
+        # get the page of the season of the show
+        logger.info('Getting the subtitle of show id %d, season %d episode %d, format %s', show_id,
+                    season, episode, format)
+        subtitles = []
+
+        # Default format is HDTV
+        sub_format = ''
+        if format is None or format.lower() == 'hdtv':
+            if resolution in ('1080i', '1080p', '720p'):
+                sub_format = resolution
+            else:
+                sub_format = 'normale'
+        else:
+            sub_format = format.lower()
+
+        params = {
+            'apikey': self.apikey,
+            'show_id': show_id,
+            'q': '%dx%02d' % (season, episode),
+            'version': sub_format
+            }
+        logger.debug(params)
+        r = self.session.get(self.server_url + 'subtitles/search', params=params, timeout=30)
+        r.raise_for_status()
+        root = etree.fromstring(r.content)
+
+        if int(root.find('data/count').text) == 0:
+            logger.warning('Subtitles not found')
+            return []
+
+        # Looking for subtitlles in first page
+        for subtitle in root.findall('data/subtitles/subtitle'):
+            if '%dx%02d' % (season, episode) in subtitle.find('name').text.lower():
```

Comment:
Also, why would you get result for other seasons/episodes if you specified it in the query?